packages/events/src/router/middleware/authorization/index.ts

```diff
@@ -202,22 +204,33 @@ export const requireAssignment: MiddlewareFunction<
     })
   }
 
+  const { user } = ctx
+
   const assignedTo = getAssignedUserFromActions(
     event.actions.filter(
       (action): action is ActionDocument =>
         action.type === ActionType.ASSIGN || action.type === ActionType.UNASSIGN
     )
   )
 
-  if (ctx.user.type === TokenUserType.SYSTEM) {
+  // System users don't require assignment
+  if (user.type === TokenUserType.SYSTEM) {
+    if (assignedTo) {
+      throw new TRPCError({
+        code: 'CONFLICT',
+        cause: 'System user can not perform action on assigned event'
+      })
+    }
```

Comment:
Just a future note: we might have to make this so integrations can manipulate record even when they are assigned. Potential conflicts etc need to be thought of..

Reply:
Good to know 👍 I was also considering whether to leave this check in or out, but decided it felt safer to put it in